src/query/stream_to_xasl.c

```diff
@@ -2413,6 +2418,26 @@ stx_build_cte_xasl_id (THREAD_ENTRY * thread_p, char *ptr, XASL_ID * cte_xasl_id
   return ptr;
 }
 
+static SQ_CACHE *
+stx_restore_sq_cache (THREAD_ENTRY * thread_p, char *ptr, int n_elements)
+{
+  SQ_CACHE *sq_cache = NULL;
+
+  if (n_elements > 0)
+    {
+      sq_cache = (SQ_CACHE *) stx_alloc_struct (thread_p, sizeof (SQ_CACHE));
+      sq_cache->sq_key_struct = stx_restore_db_value_array_extra (thread_p, ptr, n_elements, n_elements);
```

Comment:
Exception handling is required.

Reply:
I've applied your review. Please confirm.